### PR TITLE
Physical_oM: Add Construction Properties to Door and Window objects

### LINKED ISSUE #1096: Physical_oM: Add Construction properties for Window and Door objects
#### Description:

Construction properties will allow for material based workflows to function on these objects. LCA workflows, for example, will immediately benefit from these properties.

### PR DESCRIPTION
**Pull request by michaelhoehn**
### Issues addressed by this PR

Closes #1096 

### Test files

[Here is a test file](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/BHoM/Physical_oM/Physical_oM-Issue1096-AddConstructionProp/Physical_oM-%231096-AddConstructionProperty.gh?csf=1&web=1&e=Wxr7uM)

### Changelog

- Add IElementM to Door and Window
- Add Construction property to Door and Window

### CODE REVIEW

File: Physical_oM/Elements/IOpening.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
 namespace BH.oM.Physical.Elements
 {
-    public interface IOpening : IBHoMObject, IPhysical, IElement2D
+    public interface IOpening : IBHoMObject, IPhysical, IElement2D, IElementM
```
Comment (FraserGreenroyd) [inline Physical_oM/Elements/IOpening.cs:35]: I would imagine if you're making the interface the `IElementM`, you would want to enforce that all implementations of `IOpening` have a property that provides Matter based data? Either that or we should apply the `IElementM` interface only to the implementing instances.

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
Minor comment about interface placement but otherwise LGTM.

File: Physical_oM/Elements/IOpening.cs (hunk 1 above)
Comment (michaelhoehn) [reply Physical_oM/Elements/IOpening.cs]: Marking this as resolved as the latest commit removes IElementM from `IOpening` and instead implements the interface onto specific relevant objects. In this case only Door and Window have been affected.

**Comment by michaelhoehn** [comment]
@BHoMBot check installer

**Comment by michaelhoehn** [comment]
@BHoMBot check installer

**Comment by michaelhoehn** [comment]
@BHoMBot check installer

**Comment by michaelhoehn** [comment]
@BHoMBot check installer

**Review by FraserGreenroyd — APPROVED** [review]
LGTM now, thanks for the nudge @michaelhoehn

**Pull request merged**